apps/server/src/modules/oauth/uc/oauth.uc.ts

```diff
@@ -0,0 +1,137 @@
+import { Injectable } from '@nestjs/common';
+import { ILogger, Logger } from '@src/core/logger';
+import axios, { AxiosResponse } from 'axios';
+import jwtDecode from 'jwt-decode';
+import { SystemRepo } from '@shared/repo/system';
+import { UserRepo } from '@shared/repo';
+import { System, User } from '@shared/domain';
+import { FeathersJwtProvider } from '@src/modules/authorization/feathers-jwt.provider';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { Payload } from '../controller/dto/payload';
+import { OauthTokenResponse } from '../controller/dto/oauthTokenResponse';
+import { AuthorizationQuery } from '../controller/dto/authorization.query';
+import { OAuthResponse } from '../controller/dto/oauth.response';
+
+@Injectable()
+export class OauthUc {
+	private logger: ILogger;
+
+	constructor(
+		private readonly systemRepo: SystemRepo,
+		private readonly userRepo: UserRepo,
+		private readonly jwtService: FeathersJwtProvider
+	) {
+		this.logger = new Logger(OauthUc.name);
+	}
+
+	// 0- start Oauth Process
+	async startOauth(query: AuthorizationQuery, systemId: string): Promise<OAuthResponse> {
+		try {
+			// get the authorization code
+			const code: string = this.extractCode(query);
+			// get the Tokens using the authorization token
+			const queryToken: OauthTokenResponse = await this.requestToken(code, systemId);
+			// extract the uuid from the token
+			const uuid = await this.decodeToken(queryToken.id_token);
+			// get the user using the uuid
+			const user: User = await this.findUserById(uuid);
+			// create JWT for the user
+			const jwt: string = await this.getJWTForUser(user);
+			// send response back
+			const HOST = Configuration.get('HOST') as string;
+			const redirectUri = HOST.concat('/dashboard');
```

Comment:
use url.join like instead of string concat

Reply:
Which dependency / package is that? if its that one, then i have to add it to the package.json 

https://www.npmjs.com/package/url-join

is it?